components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.ts

```diff
@@ -0,0 +1,335 @@
+      default_attributes: JSON.parse(this.defaultAttrFormGroup.controls['default'].value),
+      override_attributes: JSON.parse(this.overrideAttrFormGroup.controls['override'].value)
+    };
+
+    this.store.dispatch(
+      new CreateEnvironment({environment: environment})
+    );
+  }
+
+  toDisplay(cookbookVersions: Array<DynamicGrid> = []) {
+    const current = {};
+    cookbookVersions.forEach((element) => {
+      current[element.name] =
+        `${element.operator}` + ' ' + `${element.version}`;
+    });
+    return current;
+  }
+
+  closeCreateModal(): void {
+    this.resetCreateModal();
+    this.visible = false;
+  }
+
+  private resetCreateModal(): void {
+    this.creating = false;
+    this.detailsFormGroup.reset();
+    this.defaultAttrFormGroup.reset();
+    this.overrideAttrFormGroup.reset();
+    this.showConstraint = false;
+    this.overrideTab = false;
+    this.defaultTab = false;
+    this.constraintsTab = false;
+    this.detailsTab = true;
+    this.defaultAttrFormGroup.controls.default.setValue(this.attr_value);
+    this.overrideAttrFormGroup.controls.override.setValue(this.attr_value);
+    this.defaultAttrParseError = false;
+    this.overrideAttrParseError = false;
+    this.dynamicArray = [];
+    this.loadCookbookConstraint();
+    this.conflictErrorEvent.emit(false);
+  }
+
+  private resetTabs() {
+    this.detailsTab = false;
+    this.constraintsTab = false;
+    this.defaultTab = false;
+    this.overrideTab = false;
+  }
+
+  ngOnDestroy(): void {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  private isNavigationKey(event: KeyboardEvent): boolean {
+    return event.key === 'Shift' || event.key === 'Tab';
+  }
+
+  // this is the initial value set to the component
+  public writeValue(obj: any) {
+
+    if (obj) {
+      this.data = obj;
+      // this will format it with 4 character spacing
+      this.jsonString = JSON.stringify(this.data, undefined, 4);
+    }
+  }
+
+  // registers 'fn' that will be fired wheb changes are made
+  // this is how we emit the changes back to the form
+  public registerOnChange(fn: any) {
+    this.propagateChange = fn;
+  }
+
+  public onChangeJSON(event) {
+    this.defaultAttrParseError = false;
+    this.overrideAttrParseError = false;
+    // get value from text area
+    const newValue = event.target.value;
+    this.textareaID = event.target.id;
+    try {
+        // parse it to json
+      this.data = JSON.parse(newValue);
+      if (this.textareaID === 'default') { (this.defaultAttrParseError = false); }
+      if (this.textareaID === 'override') { (this.overrideAttrParseError = false); }
+
+    } catch (ex) {
+        // set parse error if it fails
+      if (this.textareaID === 'default') { (this.defaultAttrParseError = true); }
+      if (this.textareaID === 'override') { (this.overrideAttrParseError = true); }
+
+    }
+    // update the form
+    this.propagateChange(this.data);
```

Comment:
Because `registerOnChange` is never called, this call always uses the default, which does nothing.

Reply:
updated